app/models/host/base.rb

```diff
@@ -388,7 +388,22 @@ def get_interface_scope(name, attributes, base = self.interfaces)
       end
     end
 
+    def update_attached_interfaces(name, attributes)
+      bond_interfaces.each do |bond|
+        next unless bond.children_mac_addresses.include?(attributes['macaddress'])
+        next if bond.attached_devices_identifiers.include? name
+        iface = interfaces.find_by :mac => attributes['macaddress']
```

Comment:
This forces a query to database for each bond, even if interfaces are already loaded. IMHO it would be better to use ruby's `select`:
``` ruby
interfaces.select { |iface| iface.mac == attributes['macaddress'] }
```